File: python/longestcommonprefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ""
        pre = strs[0]
        for i in xrange(1, len(strs)):
            j = 0
            while (j<min(len(strs[i]), len(pre)) and pre[j] == strs[i][j]):
                #print pre[j], strs[i][j]
                j += 1
            
            pre = pre[:j]
        return pre
                

File: cpp/longestcommonprefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";
        string res = strs[0];
        for(int i = 1; i < strs.size(); i++) {
            int index = 0;
            while(index < min(res.length(), strs[i].length()) and res[index] == strs[i][index]) index++;
            res = res.substr(0, index);
        }
        return res;
    }
};
